wordpress-mobile/WordPress-FluxC-Android
# Add lastModified to ListItemModel & PostModel

This PR is a part of post pagination project. In order to figure out whether there is a change for a post in the DB, we need to get and store the `lastModified` date of the post in both `ListItemModel` and `PostModel`. When we fetch the details of a list, we'll now fetch both the `ID` and `lastModified` date of the post. This way, when the user pulls to refresh the list, we can identify which items' contents should be updated.

As part of this PR, I also renamed `PostListModel` to `ListItemModel` to make it more generic. I was intending to make this change at a later time, but I thought this would be a good PR to combine with and we'll have less things to refactor at a later time.

To Test:
* Check the `lastModified` date with breakpoints in the example app or from connected tests. Make sure to do a fresh installation since the migration is changed.
